### PR TITLE
feat: download notifications messages body

### PR DESCRIPTION
**Pull request by sirambd**
When we display a message, we also download its body, but for the display of the body we parse it in html, while filtering the images

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt [hunk 1]
```diff
@@ -118,5 +119,32 @@ object ThreadController {
     fun deleteThreads(folderId: String, realm: MutableRealm) {
         realm.delete(getThreadsQuery(folderId, realm = realm))
     }
+
+    fun fetchIncompleteMessages(thread: Thread) {
+        RealmDatabase.mailboxContent().writeBlocking {
+            thread.messages.forEach { localMessage ->
+                if (!localMessage.fullyDownloaded) {
+                    ApiRepository.getMessage(localMessage.resource).data?.also { remoteMessage ->
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:127]: No API calls in Controllers.

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:127]: In this case, we can leave it as it is realm that drives the api call

File: app/src/main/java/com/infomaniak/mail/workers/SyncMessagesWorker.kt [hunk 2]
```diff
@@ -83,7 +85,13 @@ class SyncMessagesWorker(appContext: Context, params: WorkerParameters) : BaseCo
         mailboxInfoRealm.close()
     }
 
-    private fun Thread.showNotification(userId: Int, mailbox: Mailbox, unReadThreadsCount: Int, realm: Realm) {
+    private fun Thread.showNotification(
+        userId: Int,
+        mailbox: Mailbox,
+        unReadThreadsCount: Int,
+        realm: Realm,
+        okHttpClient: OkHttpClient
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/workers/SyncMessagesWorker.kt:93]: Remove unused `okHttpClient`.

File: app/src/main/java/com/infomaniak/mail/workers/SyncMessagesWorker.kt [hunk 3]
```diff
@@ -83,7 +85,13 @@ class SyncMessagesWorker(appContext: Context, params: WorkerParameters) : BaseCo
...
+    ) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/workers/SyncMessagesWorker.kt:94]: Revert this, `okHttpClient` is unused.

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/workers/SyncMessagesWorker.kt:94]: no it's always used

**Pull request merged**